### PR TITLE
Feature: e2e api validation

### LINKED ISSUE #15: Test E2E: Cross-cutting scenario – API endpoints validation
This scenario applies to all API endpoints exposed by the application.
For each endpoint, the following must be verified:

* Successful responses (data structure and consistency).
* Error responses (HTTP status codes, error messages, response formats).
* Proper handling of edge cases and invalid inputs.

### PR DESCRIPTION
**Pull request by christophechevalier**
Closes #15

### CODE REVIEW

File: tests/features/api/users/ModuleUsers.feature [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+  ## 503 Should return 400 when updating user without email (skipped - linagora/linid-im-api#11)
+  ## 504 Should return 400 when updating user without firstName (skipped - linagora/linid-im-api#11)
+  ## 505 Should return 400 when updating user without lastName (skipped - linagora/linid-im-api#11)
+
+  ################## Partial Update (PATCH /api/users/{id})
+  ## 601 Should return 400 for invalid JSON body on PATCH
+  ## 602 Should return 404 when patching non-existent user
+
+  ################## Delete (DELETE /api/users/{id})
+  ## 701 Should return 404 when deleting non-existent user
```
[the remark this answers is not in the document]
Comment (christophechevalier) [reply tests/features/api/users/ModuleUsers.feature]: Done

**Review by Zorin95670 — APPROVED** [review]
